w4_d1_1_intro_to_mongo/README.md

```diff
@@ -0,0 +1,169 @@
+
+### Motivation (Why?)
+
+NoSQL databases are a popular and trending database technology and it gives us a full Javascript/JSON stack.
+
+* Node - JavaScript
+* Express - JavaScript
+* jQuery - JavaScript
+* AJAX - JSON (JavaScript)
+* **Mongo - JSON (JavaScript)**
+* AngularJS - JavaScript
+
+### Analogy (What?)
+
+Imagine you're going to make a cake.
+
+SQL is like shopping for the ingredients for the recipe. At the store, you have to go to many different aisles to get each ingredient.
+
+Wouldn’t it be nicer if there was a store was organized by recipe, so you could go to one place in the store and grab everything you need from that one spot? That would be nice. And that would be like a NoSQL database.
+
+[mgoffin/Stack Overflow](http://stackoverflow.com/questions/14428069/sql-and-nosql-analogy-for-the-non-technical/14428221#14428221)
+
+> NOTE: a "model" is like a JavaScript constructor for objects we will save to our database. Models are the 'data layer' of MVC frameworks.
+
+### Key Snippets
+
+##### Enter a Node Server Console (REPL Console - Read-Eval-Print-Loop Console)
+```
+$ node
+```
+
+##### Connect a Server to Your Database
+```
+//
+// index.js or server.js
+//
+
+var mongoose = require("mongoose");
+mongoose.connect("mongodb://localhost/test");
+```
+
+##### Make a Model
+```
+//
+//  models/book.js
+//
+
+var Schema = mongoose.Schema;
+var DinosaurSchema = new Schema({
+    name: String,
+    period: String,
+    description: String,
+    height_in_inches: Integer
+});
+
+mongoose.model('Dinosaur', DinosuarSchema);
+
+//---------IN ANOTHER FILE -----------//
+var Dinosaur = require('mongoose').model('Dinosaur');
+
+```
+
+##### Query all using a Model
+
+```
+var Dinosaur = require('mongoose').model('Dinosaur');
+
+var dinosaurs = Dinosaur.find().exec(function(err, dinosaurs) {
+  console.log(dinosaurs)
+})
+```
+
+##### Save a new Model Instance to DB
+
+```
+var Dinosaur = require('mongoose').model('Dinosaur');
+
+var dinosaur = new Dinosaur({
+    name: req.body.name
+  , period: req.body.period
+  , height: req.body.height_in_in
+});
+console.log(dinosaur);
+dinosaur.save(function (err, dinosaur) {
+  console.log('dinosaur saved')
+  if (err) { return res.json(err) };
+  res.status(201).json(dinosaur)
+});
+```
+
+#Challenges
+
+### Docs & Resources
+
+* [MongoDB Shell Quick Reference](http://docs.mongodb.org/manual/reference/mongo-shell/)
+* Mongoose
+
+### Installation
+
+Install MongoDB, a popular noSQL database.
+
+* First we'll need to run brew update to update our brew packages.
+
+  ```bash
+  brew update
+  ```
+* Next we'll need to run `brew install` for **MongoDB**
+
+  ```bash
+  brew install mongodb
+  ```
+
+* Then we'll need a directory for **MongoDB** to save data.
+
+  ```bash
+  sudo mkdir -p /data/db
+  ```
+
+  * Finally we'll want to make sure we have permission to read and write to this directory.
+
+  ```bash
+  sudo chown -R $USER /data/db
+  ```
+
```

Comment:
consider adding

Side note: Curious about the new terminal commands we're using in this section? Check out their documentation in their "man" pages (short for user manual) in the terminal.  Type `man mkdir` and look for the `-p` option,  or type `man chown` and read through its documentation.